### PR TITLE
Milestone-6: Preserve data

### PR DESCRIPTION
**Pull request by MNisarAli**
### 🔍 Overview:

In this pull request, we **(@MNisarAli @lawrahkonwea @Thinus01)** implemented the ability to save and load data from JSON files in our app. 

### ♻️ Changes Made:

- [x] Refactored existing code to integrate the new functionality.
- [x] Added logic to load data from JSON files on startup.
- [x] Added error handling to account for missing JSON files.
- [x] Added logic to save data to JSON files.

### 💻 Technical Details:

- Created separate JSON files for each type of data (books, people, rentals).
- Named JSON files accordingly to improve readability and organization.

### Thank you for considering this pull request! 🙏

### CODE REVIEW

File: bin/app.rb [hunk 1]
```diff
@@ -52,6 +79,7 @@ def create_book
     author = gets.chomp
     book = Book.new(title, author)
     @books << book
+    save_books
```
Comment (julie-ify) [inline bin/app.rb:82]: - [x] Kindly fix this error. I am unable to add new books.

  juso

**Review by julie-ify — CHANGES_REQUESTED** [review]
## Hi @MNisarAli, @Thinus01 and @lawrahkonwea,

Good job so far!

## Highlights!
- Good code organization and working app ✔️
- Descriptive README ✔️ 

Hello ✋, you have done well reaching this point of the milestone 🤝 . But there are some issues that you still need to work on to go to the next project but you are almost there!

# Status: Required Changes ♻️⛔️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread and tag me `@julie-ify` if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: bin/app.rb (hunk 1 above)
Comment (MNisarAli) [reply bin/app.rb:82]: Hi @julie-ify 👋, I appreciate your time in reviewing our pull request. The `save_books` method was re-created, successfully removing the bug 🐛. Thank you 🙏.

File: bin/app.rb [hunk 2]
```diff
@@ -19,30 +20,56 @@ def list_books
...
-      person = Teacher.new(specialization, age, name: name)
+      create_teacher
     when 2
-      puts 'Enter the student classroom:'
-      classroom = gets.chomp
-      person = Student.new(Classroom.new(classroom), age, name: name)
+      create_student
+    else
+      puts 'Invalid input, Please enter a valid number!'
     end
```
Comment (tresorsawasawa) [inline bin/app.rb:47]: - [x] Good job trying to fix the issue mentioned by the previous reviewer. However, I noticed while fixing the previous error, it is occurred the issue of creating a person of `type` `Student`. Right now, only `Teacher` can be successfully created. Make sure you fix that, below is the error I get while trying to create a `Student`
Screenshot from 2023-03-30 07-44-47

**Review by tresorsawasawa — CHANGES_REQUESTED** [review]
# STATUS: REQUEST CHANGES :recycle::recycle: 

Hi @MNisarAli, @Thinus01, and @lawrahkonwea 
Great work on making the changes requested by the previous reviewer  👏👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

almost-there

## To highlight :champagne: 
- All linters are passing :heavy_check_mark: 
- Good commit history :heavy_check_mark: 
- Your design looks good :100: :heavy_check_mark: 
- Good PR description :heavy_check_mark: 

## Required changes :recycle: :recycle: 
- [x] Check the comments under the review.

## Optional suggestions

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

_Feel free to leave any questions or comments in the PR thread if something is not 100% clear. 
Please ping me [@tresorsawasawa](https://github.com/tresorsawasawa) when you comment so I can receive the notification or use slack with the same name._

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: bin/app.rb (hunk 2 above)
Comment (MNisarAli) [reply bin/app.rb:47]: Hi @tresorsawasawa  👋, I appreciate your time in reviewing our pull request. The `Student` class `initialize` method was refactored, successfully removing the bug 🐛. Thank you 🙏.

**Review by Reem-lab — APPROVED** [review]
# Hi Team,👋🏻👋🏻

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 🔥🔥💯💯


## Highlights 🔥🔥🔥🔥
- Followed Gitflow ✅
- No Linters error 🤩🔥💯
- Great working on the functionality 🔥🔥
- Descriptive PR 💯💯
- Professional README 🔥🔥



## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**